### PR TITLE
[CORE-859] `add new market` widget: msg composer

### PR DESCRIPTION
**Pull request by ttl33**
* Part 1 out 2
* Define composers for the different msgs involved in creating a new market
  * `MsgCreateOracleMarket`
  * `MsgCreatePerpetual`
  * `MsgCreateClobPair`
  * `MsgUpdateClobPair`
  * `MsgDelayMessage`
  * `MsgSubmitProposal`

### CODE REVIEW

File: v4-client-js/examples/gov_add_new_market.json
Comment (jaredvu) [inline v4-client-js/examples/gov_add_new_market.json:1]: There seems to be two example files

File: v4-client-js/src/clients/constants.ts [hunk 1]
```diff
@@ -43,6 +43,39 @@ export enum NetworkId {
...
+// x/prices
+export const TYPE_URL_MSG_CREATE_ORACLE_MARKET = '/dydxprotocol.prices.MsgCreateOracleMarket';
+
+// x/sending
+export const TYPE_URL_MSG_CREATE_TRANSFER = '/dydxprotocol.sending.MsgCreateTransfer';
+export const TYPE_URL_MSG_WITHDRAW_FROM_SUBACCOUNT = '/dydxprotocol.sending.MsgWithdrawFromSubaccount';
+export const TYPE_URL_MSG_DEPOSIT_TO_SUBACCOUNT = '/dydxprotocol.sending.MsgDepositToSubaccount';
+
+// ------------ Chain Constants ------------
+export const GOV_MODULE_ADDRESS = 'dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky';
```
Comment (jaredvu) [inline v4-client-js/src/clients/constants.ts:76]: Are these two addresses the same across mainnet, testnet, staging, and dev?

File: v4-client-js/src/clients/constants.ts [hunk 2]
```diff
@@ -43,6 +43,39 @@ export enum NetworkId {
...
+export const DELAYMSG_MODULE_ADDRESS = 'dydx1mkkvp26dngu6n8rmalaxyp3gwkjuzztq5zx6tr';
```
Comment (jaredvu) [inline v4-client-js/src/clients/constants.ts:77]: ^

File: v4-client-js/src/clients/lib/registry.ts [hunk 3]
```diff
@@ -3,26 +3,54 @@ import { defaultRegistryTypes } from '@cosmjs/stargate';
...
+    [TYPE_URL_MSG_DELAY_MESSAGE, MsgDelayMessage as GeneratedType],
+
+    // perpetuals
+    [TYPE_URL_MSG_CREATE_PERPETUAL, MsgCreatePerpetual as GeneratedType],
+
+    // prices
+    [TYPE_URL_MSG_CREATE_ORACLE_MARKET, MsgCreateOracleMarket as GeneratedType],
 
     // sending
-    ['/dydxprotocol.sending.MsgCreateTransfer', MsgCreateTransfer as GeneratedType],
```
Comment (jaredvu) [inline v4-client-js/src/clients/lib/registry.ts:23]: 🔥

File: v4-client-js/src/clients/constants.ts (hunk 1 above)
Comment (ttl33) [reply v4-client-js/src/clients/constants.ts:76]: yessir, these are the same across different deployments

File: v4-client-js/src/clients/constants.ts (hunk 2 above)
Comment (ttl33) [reply v4-client-js/src/clients/constants.ts:77]: see above

**Review by jaredvu — APPROVED** [review]

File: v4-client-js/examples/gov_add_new_market.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    composer.wrapMessageAsAny(registry, updateClobPair),
+    MOCK_DATA.delayBlocks,
+  );
+  msgs.push(createOracleMarket);
+  msgs.push(createPerpetual);
+  msgs.push(createClobPair);
+  msgs.push(delayMessage);
+
+  // x/gov.v1.MsgSubmitProposal
+  const submitProposal = composer.composeMsgSubmitProposal(
```
[the remark this answers is not in the document]
Comment (ttl33) [reply v4-client-js/examples/gov_add_new_market.ts:122]: yup exactly!

**Pull request merged**